prometheus/src/main/java/org/opensearch/sql/prometheus/functions/response/QueryRangeFunctionResponseHandle.java

```diff
@@ -5,27 +5,113 @@
 
 package org.opensearch.sql.prometheus.functions.response;
 
+import static org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants.VALUE;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensearch.sql.data.model.ExprDoubleValue;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprTimestampValue;
+import org.opensearch.sql.data.model.ExprTupleValue;
 import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
 import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.prometheus.data.constants.PrometheusFieldConstants;
 
 /**
- * Handle Prometheus response.
+ * Default implementation of QueryRangeFunctionResponseHandle.
  */
-public interface QueryRangeFunctionResponseHandle {
+public class QueryRangeFunctionResponseHandle implements PrometheusFunctionResponseHandle {
 
-  /**
-   * Return true if Prometheus  response has more result.
-   */
-  boolean hasNext();
+  private final JSONObject responseObject;
+  private Iterator<ExprValue> responseIterator;
+  private ExecutionEngine.Schema schema;
 
   /**
-   * Return Prometheus response as {@link ExprValue}. Attention, the method must been called when
-   * hasNext return true.
+   * Constructor.
+   *
+   * @param responseObject Prometheus responseObject.
    */
-  ExprValue next();
+  public QueryRangeFunctionResponseHandle(JSONObject responseObject) {
+    this.responseObject = responseObject;
+    constructIteratorAndSchema();
+  }
 
-  /**
-   * Return ExecutionEngine.Schema of the Prometheus response.
-   */
-  ExecutionEngine.Schema schema();
-}
\ No newline at end of file
+  private void constructIteratorAndSchema() {
+    List<ExprValue> result = new ArrayList<>();
+    List<ExecutionEngine.Schema.Column> columnList = new ArrayList<>();
+    if ("matrix".equals(responseObject.getString("resultType"))) {
```

Comment:
use `equalsIgnoreCase`?

Reply:
Prometheus doesn't change the case of these strings as per their api response contract. But I will extract them into static variables.